### PR TITLE
Add DropDownButtonFormField convenience class

### PR DESCRIPTION
**Pull request by jslavitz**
Fixes #16497 originally created by @ihildebrandt.

### CODE REVIEW

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -763,3 +764,70 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
...
 }
+
+/// A convenience class that wraps a [DropdownButton] in a [FormField] so as to
+/// act as a part of a [Form].
+///
+/// The [DropdownButton] widget is displayed without the underline, because it
+/// is provided by the [InputDecorator].
+///
+/// See also:
+///  * [DropdownButton], which is the underlying Widget providing the functionality
```
Comment (Hixie) [inline packages/flutter/lib/src/material/dropdown.dart]: nit: there needs to be a blank line before the line with the bullet point or it'll all get merged into one paragraph.

**Comment by Hixie** [comment]
cc @HansMuller for review

File: packages/flutter/lib/src/material/dropdown.dart [hunk 2]
```diff
@@ -763,3 +764,71 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
...
+    validator: validator,
+    builder: (FormFieldState<T> field) {
+      final InputDecoration effectiveDecoration = (decoration ?? const InputDecoration())
+        .applyDefaults(Theme.of(field.context).inputDecorationTheme);
+      return InputDecorator(
+        decoration: effectiveDecoration.copyWith(errorText: field.errorText),
+        isEmpty: value == null,
+        child: DropdownButtonHideUnderline(
+          child: DropdownButton<T>(
+              isDense: true,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown.dart]: 2-space indent for the parameters

File: packages/flutter/test/material/dropdown_test.dart [hunk 3]
```diff
@@ -237,6 +237,48 @@ void main() {
...
+              ),
+            ),
+          );
+        }
+      )
+    );
+
+    expect(value, equals('one'));
+    await tester.tap(find.text('one'));
+    await tester.pump(const Duration(seconds: 1));
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: tester.pumpAndSettle()

File: packages/flutter/test/material/dropdown_test.dart [hunk 4]
```diff
@@ -237,6 +237,48 @@ void main() {
...
+    await tester.tap(find.text('three').last);
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: tester.pumpAndSettle()

**Review by HansMuller — COMMENTED** [review]
This looks good. Just some small stuff.

**Review by HansMuller — APPROVED** [review]
Needs a few more small changes.

LGTM

**Pull request merged**